Remove sass-lint. Add stylelint.

lederer opened the pull request:
Removes `sass-lint`.

Adds `stylelint` and the baseline `stylelint-config-recommended` plugin.

Things to note:
- ~`mastarm-lint` calls `stylelint` with the `--fix` option to automatically fix whatever errors it can.~
- Apps using `mastarm` will require a `.stylelintrc` file (or equivalent).

To test:
- Add this branch as a dependency in a downstream repo
- Add a `.stylelintrc` file with some stylelint rules.
  - Even easier: 
    1. Add [stylelint-azavea](https://github.com/azavea/stylelint-azavea) as a dependency
    2. Set contents of `.stylelintrc` to `{ "extends": "stylelint-azavea" }`
- `yarn lint`

Closes #9

lederer committed:
Remove sass-lint. Add stylelint-azavea.

flibbertigibbet commented on bin/mastarm-lint:
```diff
@@ -2,4 +2,4 @@
 $CWD/../node_modules/.bin/eslint --config .eslintrc.json $FILES $@
-$CWD/../node_modules/.bin/sass-lint -c .sass-lint.yml '**/*.scss' $@ -v -q
\ No newline at end of file
+$CWD/../node_modules/.bin/stylelint '**/*.+(css|scss|sass)' --fix
```
The linter shouldn't change source by default, so leave out `--fix` here. There is a `fixlint` yarn command in ECHO Locator available for that.

flibbertigibbet reviewed (changes requested):
It looks like the new default `.stylelintrc` needs to be passed explicitly to the linter command. Running this within a project without a configuration results in a missing configuration error.

lederer committed:
Don't autofix stylelint

lederer commented:
> It looks like the new default .stylelintrc needs to be passed explicitly to the linter command. Running this within a project without a configuration results in a missing configuration error.

A .stylelintrc is required but doesn't need to be passed into the CLI. Stylelint will crawl up the file tree looking for it. Typical practice is to include one in the root of the project and let stylelint find it.

flibbertigibbet commented:
![image](https://user-images.githubusercontent.com/960264/54047579-6b78d100-41a5-11e9-88bc-87fb93e32203.png)

flibbertigibbet commented:
If the file tree it's crawling is the one for the project it's checking, it's not going to find the `mastarm` default in that path.

lederer committed:
Update README with stylelint info

lederer commented:
That's correct. The `mastarm` default is always ignored when linting within the project that's using it. This is true both if we pass a path to `.stylelintrc` in the command and if allow `stylelint` to find it instead. A project ~using~ linting with this `mastarm` fork would always require its own `.stylelintrc` in its root directory. I updated the README to clarify this.

Eg, if a path to a `.stylelintrc` is not passed into the command, and there is no `.stylelintrc` in the tree, you get the error you pasted above.

But if `--config .stylelintrc` _is_ passed in, and there is still no such `.stylelintrc`, you get this error:

![image](https://user-images.githubusercontent.com/128699/54048601-476abf00-41a8-11e9-9b02-20ad179f23fc.png)

flibbertigibbet commented:
Why not stick with the previous behavior of using mastarm's configuration as the default? Otherwise the extra dependency for `stylelint-config-recommended` doesn't seem to add much here.

lederer commented:
> Why not stick with the previous behavior of using mastarm's configuration as the default?

I would definitely prefer this, but I couldn't figure out how to make it work. :/

> Otherwise the extra dependency for stylelint-config-recommended doesn't seem to add much here.

True. I included it (1) so running `yarn lint` would work from within the mastarm project itself. This was useful while I was testing to see if it was working. (2) As a simple example of a `.stylelintrc` file.

Happy to remove it if you don't think those are sufficient reasons. 

(Or happy to leave it in if you have ideas for solving the first problem, though I don't want it to steal too many of your cycles.)

flibbertigibbet committed:
Break out linters

Separate JS and styling linters, so extra parameters may be passed through separately.

flibbertigibbet committed:
Fix linter tests

lederer committed:
Merge pull request #13 from flibbertigibbet/feature/kak/stylelint-fixes

Break out linters

flibbertigibbet reviewed (approved):
:+1:
